ethcore/verification/src/queue/kind.rs

```diff
@@ -193,17 +183,14 @@ pub mod headers {
 		type Unverified = Header;
 		type Verified = Header;
 
-		fn create(
-			input: Self::Input,
-			engine: &dyn Engine,
-			check_seal: bool
-		) -> Result<Self::Unverified, (Error, Option<Self::Input>)> {
+		fn create(input: Self::Input, engine: &dyn Engine, check_seal: bool) -> Result<Self::Unverified, Error> {
 			let res = verify_header_params(&input, engine, check_seal)
-				.and_then(|_| verify_header_time(&input));
+				.and_then(|_| Ok(verify_header_time(&input)))
+				.map_err(|error| Error::Block(BlockErrorWithData { error, data: None }))?;
 
 			match res {
 				Ok(_) => Ok(input),
-				Err(e) => Err((e, Some(input))),
+				Err(error) => Err(Error::Block(BlockErrorWithData { error, data: None })),
```

Comment:
why `None`?

Reply:
That is a great question, it is because all associated types are `Header` i.e, it doesn't  have `block bytes`.

I hope I got it right for `blocks`

Reply:
ah, right, we only care about bytes in import_block https://github.com/paritytech/parity-ethereum/blob/4e2010f2d9570487f8a5a562b6a506e260c001d5/ethcore/src/client/client.rs#L1469-L1472